src/pages-and-resources/discussions/DiscussionsSettings.test.jsx

```diff
@@ -183,22 +183,25 @@ describe('DiscussionsSettings', () => {
 
       // This is an important line that ensures the spinner has been removed - and thus our main
       // content has been loaded - prior to proceeding with our expectations.
-      await waitForElementToBeRemoved(screen.getByRole('status'));
+      await waitForElementToBeRemoved(screen.queryByRole('status'));
 
-      act(async () => {
-        userEvent.click(queryByLabelText(container, 'Select Piazza'));
-
-        userEvent.click(getByRole(container, 'button', { name: 'Next' }));
+      await userEvent.click(screen.getByLabelText('Select Piazza'));
 
-        userEvent.click(await findByRole(container, 'button', { name: 'Save' }));
+      // Have to use fireEvent.click with these Stepper buttons so that the
+      // onClick handler is triggered. (userEvent.click doesn't trigger onClick).
+      await act(async () => {
+        fireEvent.click(getByRole(container, 'button', { name: 'Next' }));
+      });
+      await act(async () => {
+        fireEvent.click(getByRole(container, 'button', { name: 'Save' }));
+      });
 
-        // This is an important line that ensures the Close button has been removed, which implies that
-        // the full screen modal has been closed following our click of Apply.  Once this has happened,
-        // then it's safe to proceed with our expectations.
-        await waitFor(() => expect(screen.queryByRole(container, 'button', { name: 'Close' })).toBeNull());
+      // This is an important line that ensures the Close button has been removed, which implies that
+      // the full screen modal has been closed following our click of Apply.  Once this has happened,
+      // then it's safe to proceed with our expectations.
+      await waitFor(() => expect(screen.queryByRole(container, 'button', { name: 'Close' })).toBeNull());
 
-        await waitFor(() => expect(window.location.pathname).toEqual(`/course/${courseId}/pages-and-resources`));
-      });
+      await waitFor(() => expect(window.location.pathname).toEqual(`/course/${courseId}/pages-and-resources`));
```

Comment:
This test and [another test below](https://github.com/openedx/frontend-app-course-authoring/pull/1062/files#diff-a98eb42e6f50767c3eed8ff0c6cb628fedb65953de38129770e3a49fae4d090aR358) are failing with a warning I can't work out:

> No routes matched location "/course/course-v1:edX+TestX+Test_Course/pages-and-resources"

That route is [defined correctly](https://github.com/open-craft/frontend-app-course-authoring/blob/jill/fix-test-warnings/src/CourseAuthoringRoutes.jsx#L66-L69) and works when I run the MFE.

Help?

Reply:
@pomegranited I looked into this, for some reason the `window.location` doesnt get updated, however when you use the `useLocation` it gets updated as expected. So I created this PR that seems to fix the issue. https://github.com/open-craft/frontend-app-course-authoring/pull/36

Reply:
Thank you so much @yusuf-musleh , that did the trick!